pmm/metrics.go

```diff
@@ -73,47 +88,61 @@ func (a *Admin) AddMongoDBMetrics(uri, cluster string) error {
 		Service: &srv,
 	}
 	if _, err := a.consulAPI.Catalog().Register(&reg, nil); err != nil {
-		return err
+		return nil, err
 	}
 
 	// Add info to Consul KV.
-	d := &consul.KVPair{Key: fmt.Sprintf("%s/%s/dsn", a.Config.ClientName, serviceID),
-		Value: []byte(SanitizeDSN(uri))}
-	a.consulAPI.KV().Put(d, nil)
+	if len(m.KV()) > 0 {
+		for i, v := range m.KV() {
+			d := &consul.KVPair{
+				Key:   fmt.Sprintf("%s/%s/%s", a.Config.ClientName, serviceID, i),
+				Value: v,
+			}
+			a.consulAPI.KV().Put(d, nil)
```

Comment:
Error return value of `(*github.com/percona/pmm-client/vendor/github.com/hashicorp/consul/api.KV).Put` is not checked